### PR TITLE
New onboarding

### PR DESCRIPTION
**Pull request by elie222**
## Summary by CodeRabbit

* **New Features**
  * Introduced a multi-step onboarding flow for email accounts, including steps for introduction, role selection, label/category setup, and digest preferences.
  * Added new components for onboarding UI, such as benefit cards, icon circles, scrollable containers, and navigational buttons.
  * Implemented AI-powered persona analysis based on user emails.
  * Enhanced survey with richer user role data, including descriptions and icons.

* **Improvements**
  * Digest settings forms now support conditional display of save buttons and improved layout/text.
  * Onboarding navigation and layout adapt dynamically based on the current step and path.

* **Removals**
  * Removed legacy onboarding steps and components, including bulk unsubscriber, cold email blocker, AI email assistant, onboarding finish, next button, and stepper UI.

* **Tests**
  * Added comprehensive test suite for AI persona analysis, covering multiple user roles and edge cases.

* **Documentation**
  * Updated validation schemas and type definitions for onboarding actions.

### CODE REVIEW

File: apps/web/app/(app)/[emailAccountId]/onboarding/page.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+export default async function OnboardingPage(props: {
+  params: Promise<{ emailAccountId: string }>;
+  searchParams: Promise<{ step?: string; force?: string }>;
+}) {
+  const { emailAccountId } = await props.params;
+  const searchParams = await props.searchParams;
+  const session = await auth();
+
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/onboarding/page.tsx:25]: @coderabbitai
no, next 15 does pass these as promises
